test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,287 @@ def forward(self, x):
             module_entry = per_channel_info["per_channel_status"][key]
             self.assertEqual(module_entry["per_channel_supported"], True)
             self.assertEqual(module_entry["per_channel_used"], False)
+
+
+"""
+Partition on Domain / Things to Test
+
+- All zero tensor
+- Multiple tensor dimensions
+- All of the outward facing functions
+- Epoch min max are correctly updating
+- Batch range is correctly averaging as expected
+- Reset for each epoch is correctly resetting the values
+
+Partition on Output
+- the calcuation of the ratio is occurring correctly
+
+"""
+
+
+class TestModelReportObserver(QuantizationTestCase):
```

Comment:
wondering if we can simplify the tests a bit, in general it's better to test for the functionality instead of implementation, and the functionality here is mainly to get the correct ratio I think

Reply:
I agree overall, (including some of the ones you highlighted below). However since technically updating some of these values is the job of the forward() function, I thought overall it wouldn't hurt to add a couple of quick checks with unique edge test cases to ensure it was working properly overall.

Reply:
sure, sounds good to me to check edge cases